### PR TITLE
hidden services: add auth for clients

### PR DESCRIPTION
**Pull request by fabianfrz**
follow up for #307 
make it possible to authenticate as a client

### CODE REVIEW

File: security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml [hunk 1]
```diff
@@ -99,5 +99,23 @@
...
     </dns_map_hosts>
+    <client_authentications>
+      <client_auth type="ArrayField">
+        <enabled type="BooleanField">
+          <default>1</default>
+          <Required>Y</Required>
+        </enabled>
+        <onion_service type="TextField">
+          <Required>Y</Required>
+          <default>exampleexample23.onion</default>
```
Comment (fichtner) [inline security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml]: the default is weird here

Comment (fabianfrz) [reply security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml]: better idea? it has to have an exact length and needs to consist of characters a subset of numbers.

Comment (fichtner) [inline security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml]: default is a real world sane value, you're using it as a placeholder / indicator. rather have no default than something like that, but everything is set to Required so it cannot be modelled as empty...

Comment (fabianfrz) [reply security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml]: it must be required as it would not work without the field being set.

Comment (fabianfrz) [reply security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml]: this is the value of my dev VM:
kmlnhay2lcmahggk.onion

File: security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml [hunk 2]
```diff
@@ -99,5 +99,23 @@
...
+          <mask>/^[a-z2-7]{16}\.onion$/i</mask>
+        </onion_service>
+        <auth_cookie type="TextField">
+          <Required>Y</Required>
+          <default>0</default>
```
Comment (fabianfrz) [inline security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml]: just for your information - this is stupid too

File: security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml (hunk 1 above)
Comment (fichtner) [reply security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml]: this is more of a Required set of data than individual Required values, something we can't solve today. fine.

File: security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml (hunk 2 above)
Comment (fichtner) [reply security/tor/src/opnsense/mvc/app/models/OPNsense/Tor/General.xml]: :D

**Comment by fabianfrz** [comment]
@fichtner how cool is that - GitHub seems to sign commits now ;)
Maybe I should sign my stuff using my GPG smartcard ;)

**Pull request merged**